types/types.go

```diff
@@ -111,3 +113,32 @@ func UnpackWarpMessage(unsignedMsgBytes []byte) (*avalancheWarp.UnsignedMessage,
 	}
 	return unsignedMsg, nil
 }
+
+func LogsToBlocks(logs []types.Log) ([]*WarpBlockInfo, error) {
+	slices.SortFunc(logs, func(a, b types.Log) int {
+		if a.BlockNumber < b.BlockNumber {
+			return -1
+		}
+		if a.BlockNumber > b.BlockNumber {
+			return 1
+		}
+		return 0
+	})
+
+	var blocks []*WarpBlockInfo
+	for _, log := range logs {
+		// If this log's block hasn't been seen yet, append a new entry to the slice
+		if len(blocks) == 0 || blocks[len(blocks)-1].BlockNumber != log.BlockNumber {
+			blocks = append(blocks, &WarpBlockInfo{
+				BlockNumber: log.BlockNumber,
+				Messages:    make([]*WarpMessageInfo, 0),
+			})
+		}
```

Comment:
This step seems really awkward. Could we do something like, create a `map[int][]*WarpMessageInfo`. Then for each iteration we just append the `warpMessageInfo` entry at that block number. Then, when we have the full map, we can iterate over the entries creating a `[]*WarpBlockInfo`, then sort.

Reply:
I just opted for a full simplification where this function returns the map itself and then we no longer need to do sorting or anything. I think it's a lot cleaner and requires fewer `append` allocations except for re-appending to the `WarpMessageInfo.Messages` slice